File: 75-sort-colors/sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int l=0;
        int m=0;
        int h=nums.length-1;

        while (m<=h){
            if(nums[m]==0){
                //put 0 to start
                int temp=nums[l];
                nums[l]=nums[m];
                nums[m]=temp;
                l++;
                m++;
            }
            else if(nums[m]==1){
                //put 1 in middle (already true)
                m++;
            }
            else{
                //put 2 in end
                int temp = nums[m];
                nums[m]=nums[h];
                nums[h]=temp;
                h--;
            }
        }
    }
}

File: 75-sort-colors/sort-colors.py
class Solution(object):
    def sortColors(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        low,mid,high=0,0,len(nums)-1
        while mid <= high:
            if nums[mid]==0: #put 0 in start
                nums[low],nums[mid]=nums[mid],nums[low]
                low+=1
                mid+=1
            elif nums[mid]==1: #put 1 in middle(alr present due to prev iteration)
                mid+=1
            else: #put 2 in end
                nums[mid],nums[high]=nums[high],nums[mid]
                high-=1